client/utils/currency/index.js

```diff
@@ -28,13 +28,13 @@ const currencyNames = {
 export const formatCurrencyName = ( currencyCode ) =>
 	currencyNames[ currencyCode.toLowerCase() ] || currencyCode.toUpperCase();
 
+/* eslint-disable valid-jsdoc */
 /**
  * Gets wc-admin Currency for the given currency code
  *
  * @param {string} currencyCode Currency code
  * @param {string} baseCurrencyCode Base Currency code to override decimal and thousand separators
  *
- * @return {Currency|null} Currency object
```

Comment:
Sadly, the [Currency factory](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/currency/src/utils.tsx#L65) is just typed as a generic JS [Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) in the package.
If we want to eliminate the custom types we had defined in `declarations.d.ts`, we need to let TS infer the type on its own. Or I could update the `CurrencyInterface` we defined there and use it as the return type here, but this seems like a duplicate effort given that everything else is typed already in the package, and also CTRL+click would make the IDEA navigate to our custom type instead of going inside the package directly.

Anyway, that's why I also had to temporarily disable the ESLint rule `valid-jsdoc`, which requires a `@return` annotation.